services/pseudo_controller/src/main/java/com/futurewei/alcor/pseudo_controller/pseudo_controller.java

```diff
@@ -0,0 +1,387 @@
+/*
+This is the code for the test controller, for testing the reactions between the Network Configuration manager and
+the ACA.
+
+Params:
+1. Number of ports to generate to each aca node
+2. IP of aca_node_one
+3. IP of aca_node_two
+4. IP of the GRPC call
+5. Port of the GRPC call
+6. User name of aca_nodes
+7. Password of aca_nodes
+*/
+package com.futurewei.alcor.pseudo_controller;
+
+import com.futurewei.alcor.schema.*;
+import com.jcraft.jsch.Channel;
+import com.jcraft.jsch.ChannelExec;
+import com.jcraft.jsch.JSch;
+import com.jcraft.jsch.Session;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.stub.StreamObserver;
+
+import java.io.InputStream;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+
+public class pseudo_controller {
+
+    static String aca_node_one_ip = "ip_one";
+    static String aca_node_two_ip = "ip_two";
+    static String ncm_ip = "ip_three";
+    static int ncm_port = 123;
+    static String user_name = "root";
+    static String password = "abcdefg";
+    static int ports_to_generate_on_each_aca_node = 1;
+    static String docker_ps_cmd = "docker ps";
+    static String vpc_id_1 = "2b08a5bc-b718-11ea-b3de-111111111112";
+    static String port_ip_template = "11111111-b718-11ea-b3de-";
+    static String subnet_id_1 = "27330ae4-b718-11ea-b3df-111111111113";
+    static String ips_ports_ip_prefix = "10";
+    static String mac_port_prefix = "6c:dd:ee:";
+    static SortedMap<String, String> ip_mac_map = new TreeMap<>();
+    static Vector<String> aca_node_one_commands = new Vector<>();
+    static Vector<String> aca_node_two_commands = new Vector<>();
+    static SortedMap<String, String> port_ip_to_host_ip_map = new TreeMap<>();
+    static SortedMap<String, String> port_ip_to_id_map = new TreeMap<>();
+    static SortedMap<String, String> port_ip_to_container_name = new TreeMap<>();
+    static Vector<String> node_one_port_ips = new Vector<>();
+    static Vector<String> node_two_port_ips = new Vector<>();
+
+    public static void main(String[] args) {
+        System.out.println("Start of the test controller");
+        if (args.length == 7) {
+            System.out.println("User passed in params and we need to read them.");
```

Comment:
@zzxgzgz do we plan to use TC for interactive mode or just be called by script? If we plan to be used  for interactive mode, we better prompt the usage and verify user's input.

Reply:
I think we are just calling the TC, there's no interaction at the moment.